packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py

```diff
@@ -126,136 +150,186 @@ def __init__(self, **data: Any):
-            # this is a link from storage
-            value = await port_utils.pull_file_from_store(
-                user_id=self._node_ports.user_id,
-                key=self.key,
-                fileToKeyMap=self.file_to_key_map,
-                value=self.value,
-            )
-        elif isinstance(self.value, DownloadLink):
-            # this is a downloadable link
-            value = await port_utils.pull_file_from_download_link(
-                key=self.key,
-                fileToKeyMap=self.file_to_key_map,
-                value=self.value,
-            )
-        else:
-            # this is directly the value
-            value = self.value
-        # don't atempt conversion of None it fails
-        if value is None:
-            return None
+        async def _evaluate():
+            if self.value is None:
+                return None
+
+            if isinstance(self.value, PortLink):
+                # this is a link to another node
+                other_port_concretevalue: Optional[
+                    ItemConcreteValue
+                ] = await port_utils.get_value_from_link(
+                    # pylint: disable=protected-access
+                    key=self.key,
+                    value=self.value,
+                    fileToKeyMap=self.file_to_key_map,
+                    node_port_creator=self._node_ports._node_ports_creator_cb,
+                )
+                value = other_port_concretevalue
 
-        return self._py_value_converter(value)
+            elif isinstance(self.value, FileLink):
+                # this is a link from storage
+                path_concrete_value: Path = await port_utils.pull_file_from_store(
+                    user_id=self._node_ports.user_id,
+                    key=self.key,
+                    fileToKeyMap=self.file_to_key_map,
+                    value=self.value,
+                )
+                value = path_concrete_value
+
+            elif isinstance(self.value, DownloadLink):
+                # this is a downloadable link
+                path_concrete_value: Path = (
+                    await port_utils.pull_file_from_download_link(
+                        key=self.key,
+                        fileToKeyMap=self.file_to_key_map,
+                        value=self.value,
+                    )
+                )
+                value = path_concrete_value
 
-    async def _set(self, new_value: ItemConcreteValue) -> None:
+            else:
+                # otherwise, this is a BasicValueTypes
+                value = self.value
+
+            # don't atempt conversion of None it fails
+            if value is None:
+                return None
+
+            concrete_value = self._py_value_converter(value)
+            return concrete_value
+
+        if not self._concrete_value:
+            self._concrete_value = await _evaluate()
+
+        return self._concrete_value
+
+    async def _set(self, new_concrete_value: ItemConcreteValue) -> None:
```

Comment:
very minor: not sure I understand the added value of renaming this. Is the type not already sufficient?